src/devices/Mcp25xxx/Mcp25xxx.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using Iot.Device.Mcp25xxx.Register;
+
+namespace Iot.Device.Mcp25xxx
+{
+    /// <summary>
+    /// A general purpose driver for the Microchip MCP25 CAN controller device family.
+    /// </summary>
+    public abstract class Mcp25xxx : IDisposable
+    {
+        internal IGpioController _masterGpioController;
+        private SpiDevice _spiDevice;
+        private readonly int _reset;
+        private readonly int _tx0rts;
+        private readonly int _tx1rts;
+        private readonly int _tx2rts;
+        private readonly int _interrupt;
+        private readonly int _rx0bf;
+        private readonly int _rx1bf;
+        private readonly int _clkout;
+
+        /// <summary>
+        /// A general purpose driver for the Microchip MCP25 CAN controller device family.
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="reset">The output pin number that is connected to Reset.</param>
+        /// <param name="tx0rts">The output pin number that is connected to Tx0RTS.</param>
+        /// <param name="tx1rts">The output pin number that is connected to Tx1RTS.</param>
+        /// <param name="tx2rts">The output pin number that is connected to Tx2RTS.</param>
+        /// <param name="interrupt">The input pin number that is connected to INT.</param>
+        /// <param name="rx0bf">The input pin number that is connected to Rx0BF.</param>
+        /// <param name="rx1bf">The input pin number that is connected to Rx1BF.</param>
+        /// <param name="clkout">The input pin number that is connected to CLKOUT.</param>
+        /// <param name="masterGpioController">
+        /// The GPIO controller for defined external pins. If not specified, the default controller will be used.
+        /// </param>
+        public Mcp25xxx(
+            SpiDevice spiDevice,
+            int reset = -1,
+            int tx0rts = -1,
+            int tx1rts = -1,
+            int tx2rts = -1,
+            int interrupt = -1,
+            int rx0bf = -1,
+            int rx1bf = -1,
+            int clkout = -1,
+            IGpioController masterGpioController = null)
```

Comment:
just `gpioController`

Reply:
Yeah, that probably is better.  I was using from Mcp23xxx code.  I'll update.